spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/RabbitConnectionFactoryBean.java

```diff
@@ -297,7 +300,12 @@ protected ConnectionFactory createInstance() throws Exception {
 	 */
 	protected void setUpSSL() throws Exception {
 		if (this.sslPropertiesLocation == null) {
-			this.connectionFactory.useSslProtocol();
+			if (this.sslAlgorithmSet) {
+				this.connectionFactory.useSslProtocol(this.sslAlgorithm);
+			}
+			else {
+				this.connectionFactory.useSslProtocol();
```

Comment:
👍

> On 1/6/2015, at 16:22, Gary Russell <notifications@github.com> wrote:
> 
> In spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/RabbitConnectionFactoryBean.java:
> 
> > @@ -297,7 +300,12 @@ protected ConnectionFactory createInstance() throws Exception {
> >  	 */
> >  	protected void setUpSSL() throws Exception {
> >  		if (this.sslPropertiesLocation == null) {
> > -			this.connectionFactory.useSslProtocol();
> > +			if (this.sslAlgorithmSet) {
> > +				this.connectionFactory.useSslProtocol(this.sslAlgorithm);
> > +			}
> > +			else {
> > +				this.connectionFactory.useSslProtocol();
> I thought it better to provide an option to use the default supplied by the rabbit client (in case it changes in future).
> 
> i.e. if you don't specify it in Spring configuration, you get the rabbitmq client default.
> 
> I don't want to get into the business of synchronizing our default with the rabbit client's.
> 
> —
> Reply to this email directly or view it on GitHub.
>